src/Core/Ripgrep.re

```diff
@@ -148,30 +192,60 @@ let process = (rgPath, args, callback, completedCallback) => {
       },
     );
 
-  () => {
+  let dispose = () => {
     Log.info("Ripgrep session complete.");
-    dispose1();
-    dispose2();
-    switch (dispose3^) {
+    disposeOnData();
+    disposeOnClose();
+    switch (disposeTick^) {
     | Some(v) => v()
     | None => ()
     };
-    cp.kill(Sys.sigkill);
+    childProcess.kill(Sys.sigkill);
   };
+
+  dispose;
 };
 
+
 /**
    Search through files of the directory passed in and sort the results in
    order of the last time they were accessed, alternative sort order includes
    path, modified, created
  */
 let search = (path, workingDirectory, callback, completedCallback) => {
+  let dedup = {
+    let seen = Hashtbl.create(1000);
+
+    List.filter(str => {
+      switch (Hashtbl.find_opt(seen, str)) {
+      | Some(_) => false
+      | None =>
+        Hashtbl.add(seen, str, true);
+        true;
+      };
+    });
+  };
+
   process(
     path,
     [|"--smart-case", "--files", "--", workingDirectory|],
-    callback,
+    items => items |> dedup |> callback,
+    completedCallback,
+  );
+};
+
+let findInFiles = (path, workingDirectory, query, callback, completedCallback) => {
+  process(
+    path,
+    [|"--smart-case", "--json", "--", query, workingDirectory|],
```

Comment:
I'm assuming this is duplicating the flags from the file opening search, but may make sense to have `--hidden` in both searches. Otherwise, we aren't searching any hidden directories.

This is especially noticable in repos that contain only/lots of hidden files like a dotfiles repo.

(Not exactly urgent or hard, but just something I bump up against occasionally in the existing quick open.)

Reply:
I have no strong opinion on this, but we might ask "What would vscode do?"

Reply:
VSCode by deafult does seem to search in hidden folders. Testing in the Oni2 dir with `esy__cache_install` returns 3 files in VSCode vs the 1 file in Oni2. The results are the same for `azure-pipelines.yml` but the Oni2 searches misses the two `.ci/` files.

Reply:
Then I guess we're doing that :)